src/api/java/Context.java

```diff
@@ -4052,15 +4048,7 @@ public void dispose()
         m_stringSort = null;
 
         synchronized (creation_lock) {
```

Comment:
Why wouldn't it be public? With Autocloseable, the client code can do the following:

```java
try (Context c = Context.mkContext()) {
   // do stuff here
}
```

If we want to definitely avoid segfaults on second "close" we can add a boolean variable `isClosed` and test for that instead.
I think this is much better than using non-final fields.